katello/katello-change-hostname

```diff
@@ -263,6 +263,13 @@ unless @foreman_proxy_content
   # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
   # For the same reason, we accept exit code 65 here.
   hammer_cmd("capsule update --id #{proxy_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
+
+  STDOUT.puts "Updating installation media paths"
+  installation_media_ids = hammer_cmd("medium list | grep #{old_hostname} | awk '{ print $1 }'").split("\n")
+  installation_media_paths =  hammer_cmd("medium list | grep #{old_hostname} | awk '{ print $5 }'").split("\n")
```

Comment:
`hammer medium list` will shorten the path if it is too long. e.g.:

    9  | Default_Organization/Library/Red_Hat_Server/Red_Hat_Enterprise_Linux_7_Server... | http://sat.example.com/pulp/repos/Default_Organization/Library/con...        

If you use that output to update the path, save the shortened path, instead of the real one, breaking it :(

Reply:
I think something like this might be fixing it:

```ruby
old_mediums = JSON.parse(hammer_cmd("--output json medium list --search #{@old_hostname}"))
old_mediums.each do |medium|
  new_path = medium['Path'].sub(@old_hostname, @new_hostname)
  hammer_cmd("medium update --id #{medium['Id']} --path #{new_path}")
end
```

(untested)

Reply:
I propose the following (tested) patch on top of this PR:

```diff
--- a/katello-packaging/katello/katello-change-hostname
+++ b/katello-packaging/katello/katello-change-hostname
@@ -6,6 +6,7 @@ require "rubygems"
 require 'yaml'
 require 'shellwords'
 require "json"
+require 'uri'
 
 raise 'Must run as root' unless Process.uid == 0
 
@@ -275,10 +276,12 @@ unless @foreman_proxy_content
   hammer_cmd("capsule update --id #{proxy_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
 
   STDOUT.puts "Updating installation media paths"
-  installation_media_ids = hammer_cmd("medium list | grep #{@old_hostname} | awk '{ print $1 }'").split("\n")
-  installation_media_paths =  hammer_cmd("medium list | grep #{@old_hostname} | awk '{ print $5 }'").split("\n")
-  installation_media_ids.zip(installation_media_paths).each do |id, path|
-    hammer_cmd("medium update --id #{id} --path #{path.sub! @old_hostname, @new_hostname}")
+  old_media = JSON.parse(hammer_cmd("--output json medium list --search #{@old_hostname}"))
+  old_media.each do |medium|
+    new_path = URI.parse(medium['Path'])
+    new_path.hostname = @new_hostname
+    new_path = new_path.to_s
+    hammer_cmd("medium update --id #{medium['Id']} --path #{new_path}")
   end
 end

 ```